docs/tutorials/basic/symbol.md

```diff
@@ -1,23 +1,73 @@
 # Symbol - Neural network graphs and auto-differentiation
 
-Besides the tensor computation interface `NDArray`, another main object in MXNet
-is the `Symbol` provided by `mxnet.symbol`, or `mxnet.sym` for short. A symbol
-represents a multi-output symbolic expression. They are composited by operators,
-such as simple matrix operations (e.g. "+"), or a neural network layer
-(e.g. convolution layer). An operator can take several input variables, produce
-more than one output variables, and have internal state variables. A variable
-can be either free, which we can bind with value later, or an output of another
-symbol.
-
-## Symbol Composition
+In a [previous tutorial](ndarray.md), we introduced `NDArray`,
+the basic data structure for manipulating data in MXNet.
+And just using NDArray by itself, we can execute a wide range of mathematical operations.
+In fact, we could define and update a full neural network just by using `NDArray`.
+`NDArray` allows you to write programs for scientific computation
+in an imperative fashion using, making full use of the native control of any front end language.
+So you might wonder, why don't we just use `NDArray` for all computation?
+
+MXNet also provides the Symbol API, an interface for symbolic programming.
+With symbolic programing, rather than executing operations step by step,
+we first define a *computation graph*.
+This graph contains placeholders for inputs and designated outputs.
+We can then compile the graph, yielding a function
+that can be bound to real data and run.
+MXNet's Symbol API is similar to the network configurations
+used by [Caffe](http://caffe.berkeleyvision.org/)
+and [CXXNet](https://github.com/dmlc/cxxnet)
+and the symbolic programming in [Theano](http://deeplearning.net/software/theano/).
+
+There are a few reasons for running symbolic operations.
+First, building symbolic computation graphs
+makes it easy to automatically calculate derivatives.
+Nearly all neural networks are trained by defining a model
+with a set of parameters and some objective function.
+The very act of *learning* in neural networks corresponds
+to taking derivatives (the *gradient*) of the objective
+with respect to the parameters.
+By updating the parameters in the direction indicated by the gradient,
+we improve performance with respect to the objective.
+Because taking derivatives is so integral to training neural networks,
+the ability to compute them automatically and efficiently is a tremendous advantage.
+
+Another advantage conferred by symbolic aproach is that
+we can optimize our functions before using them.
+For example, we execute mathematical computations in imperative fashion,
+we don't know at the time that we run each operation,
+which values will be needed later on.
+But with symbolic programming, we declare the required outputs in advance.
+This means that we can recycle memory allocated in intermediate steps,
+as by performing operations in place.
```

Comment:
How about a brief sentence saying symbolic API uses less memory for the same network and refer users to design notes for how and why.